GitHubTeams.ps1

```diff
@@ -691,219 +1629,328 @@ filter Set-GitHubTeam
-            Position = 2,
             ParameterSetName='TeamName')]
         [ValidateNotNullOrEmpty()]
-        [string] $TeamName,
+        [String] $TeamName,
 
         [Parameter(
             Mandatory,
             ValueFromPipelineByPropertyName,
             ParameterSetName='TeamSlug')]
-        [ValidateNotNullOrEmpty()]
         [string] $TeamSlug,
 
         [Parameter(
             Mandatory,
-            ValueFromPipelineByPropertyName,
-            Position = 3)]
-        [ValidateNotNullOrEmpty()]
-        [string] $NewTeamName,
+            ValueFromPipelineByPropertyName)]
+        [Alias('ProjectId')]
+        [int64] $Project,
 
-        [switch] $PassThru,
+        [ValidateScript({ -not $_ -or @('Admin', 'Read', 'Write') -contains $_ })]
+        [string] $Permission,
 
         [string] $AccessToken
     )
 
     Write-InvocationLog
 
-    if (-not $PSBoundParameters.ContainsKey('TeamSlug'))
+    $telemetryProperties = @{
+        OrganizationName = (Get-PiiSafeString -PlainText $OrganizationName)
+        TeamName = (Get-PiiSafeString -PlainText $TeamName)
+        TeamSlug = (Get-PiiSafeString -PlainText $TeamSlug)
+        Project = (Get-PiiSafeString -PlainText $Project)
+        Permission = (Get-PiiSafeString -PlainText $Permission)
+    }
+
+    if ($PSCmdlet.ParameterSetName -eq 'TeamName')
     {
-        $team = Get-GitHubTeam -OrganizationName $OrganizationName -TeamName $TeamName -AccessToken:$AccessToken
+        $getGitHubTeamParms = @{
+            OrganizationName = $OrganizationName
+            TeamName = $TeamName
+        }
+
+        if ($PSBoundParameters.ContainsKey('AccessToken'))
+        {
+            $getGitHubTeamParms['AccessToken'] = $AccessToken
+        }
+
+        $team = Get-GitHubTeam @getGitHubTeamParms
         $TeamSlug = $team.slug
     }
 
+    $body = $null
+    if ($Permission)
+    {
+        $hashBody = @{
+            permission = $Permission.ToLowerInvariant()
+        }
+
+        $body = $hashBody | ConvertTo-Json
+    }
+
+    if (-not $PSCmdlet.ShouldProcess($TeamSlug, 'Set GitHub Team Project Permission'))
+    {
+        return
+    }
+
     $params = @{
-        OrganizationName = $OrganizationName
-        TeamSlug = $TeamSlug
-        TeamName = $NewTeamName
-        PassThru = (Resolve-ParameterWithDefaultConfigurationValue -Name PassThru -ConfigValueName DefaultPassThru)
+        UriFragment = "orgs/$OrganizationName/teams/$TeamSlug/projects/$Project"
+        Body = $body
+        Method = 'Put'
+        Description = "Setting project permission of team $TeamSlug"
         AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+        AcceptHeader = $script:inertiaAcceptHeader
     }
 
-    return Set-GitHubTeam @params
+    Invoke-GHRestMethod @params | Out-Null
 }
 
-filter Remove-GitHubTeam
+filter Set-GitHubTeamRepositoryPermission
```

Comment:
Like above, the object being changed isn't the team.  In this case it's the repository.  You're also not setting the list of supported teams (which would imply that this removes teams too), but rather only _adding_ a team with its permissions.  I'd suggest that this be a command in `GitHubRepositories.ps1` and be called `Add-GitHubRepositoryTeam`.  Given that the endpoint can _both_ add _and_ update a single team's permission, it would be relevant to add an alias of `Set-GitHubRepositoryTeam` as well.